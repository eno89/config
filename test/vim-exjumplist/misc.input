" Test: vim-exjumplist misc
"
" NB: There is no way to reset jumplist easily,
"     so that all tests in this file may have side effects.

runtime! plugin/exjumplist.vim
call vspec#hint({'scope': 'exjumplist#_scope()', 'sid': 'exjumplist#_sid()'})
function s:describe__s_jumplist_info__at_first()  "{{{1
  " FIXME: NIY
endfunction




function s:describe__s_jumplist_info__at_last()  "{{{1
  " FIXME: NIY
endfunction




function s:describe__s_jumplist_info__at_middle()  "{{{1
  " Set up jumplist - 2+ / 1
  silent help
  normal! 10G
  normal! 100G
  execute 'normal!' "\<C-o>"

  It should return valid values

  let [o1, n1] = Call('s:jumplist_info')
  Should o1 >= 2
  Should n1 >= 1
  Should o1 + n1 >= 3

  It should return valid values after <C-o>

  execute 'normal!' "\<C-o>"
  let [o2, n2] = Call('s:jumplist_info')
  Should o2 + 1 == o1
  Should n2 - 1 == n1

  It should return valid values after <C-i>

  execute 'normal' "\<Plug>(exjumplist-%-next-position)"
  let [o3, n3] = Call('s:jumplist_info')
  Should o3 - 1 == o2
  Should n3 + 1 == n2
  Should o3 == o1
  Should n3 == n1
endfunction




" __END__  "{{{1
" vim: filetype=vim foldmethod=marker
