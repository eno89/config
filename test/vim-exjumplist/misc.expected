==== s_jumplist_info at first

==== s_jumplist_info at last

==== s_jumplist_info at middle
---- It should return valid values 
...
---- It should return valid values after <C-o> 
..
---- It should return valid values after <C-i> 
....



**** Result ****


9 examples, 0 failures
